src/Time/Slab.cpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/Slab.hpp"
+
+#include <pup.h>
+
+#include "Time/Time.hpp"
+
+// These things are here to avoid problems with Time possibly being
+// incomplete in the header.
+
+Time Slab::start() const noexcept { return Time(*this, 0); }
+
+Time Slab::end() const noexcept { return Time(*this, 1); }
+
+TimeDelta Slab::duration() const noexcept { return TimeDelta(*this, 1); }
+
+Slab Slab::advance_towards(const TimeDelta& dt) const noexcept {
+  ASSERT(dt.is_positive() or (-dt).is_positive(),
+         "Can't advance along a zero time vector");
+  return dt.is_positive() ? advance() : retreat();
+}
+
+void Slab::pup(PUP::er& p) noexcept {
+  p | start_;
+  p | end_;
+}
+
+std::ostream& operator<<(std::ostream& os, const Slab& s) noexcept {
+  return os << "Slab[" << s.start().value() << "," << s.end().value() << "]";
+}
+
+namespace std {
+size_t hash<Slab>::operator()(const Slab& s) const noexcept {
+  return hash<double>()(s.start().value()) ^ hash<double>()(s.end().value());
```

Comment:
I think I agree with you, I'd rather use the symbols for bitwise operators (Nils raised the question in code review by left it as soliciting your opinion, not a requested change)